pkg/network/vmispec/interface.go

```diff
@@ -128,3 +118,13 @@ func LookupInterfaceByName(ifaces []v1.Interface, name string) *v1.Interface {
 	}
 	return nil
 }
+
+func IndexInterfaceStatusByName(interfaces []v1.VirtualMachineInstanceNetworkInterface, p func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool) map[string]v1.VirtualMachineInstanceNetworkInterface {
```

Comment:
I would suggest to rename `p` into something more descriptive, probably `doesMatch` or `isMatching`. WDYT?